src/components/calculator/Calculator.jsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from "react";
+import Display from "./Display";
+import Button from "./Button";
+
+function Calculator() {
+    const [number, setNumber] = useState(0);
+    const [oldNumber, setOldNumber] = useState(0);
+    const [operator, setOperator] = useState();
+    const [flag, setFlag] = useState(false);
+
+    function inputNumber(e) {
+        parseInt(number) === 0 || flag ? setNumber(e.target.value) : setNumber(number + e.target.value);
+    }
+
+    const clear = () => {
+        setNumber(0);
+        setOldNumber(0);
+    }
+
+    function changeSignal() {
+        setNumber(number * -1);
+    }
+
+    function getPercentage() {
+        setNumber(number / 100);
+    }
+
+    function handleOperator(e) {
+        let op = e.target.value;
```

Comment:
Uma outra forma de obteres um valor dentro de um objeto:
```suggestion
        const { target: value } = e;
```

Reply:
Is that ok even tho the value changes with any other operator ?

Reply:
Eu tentei e penso que não funciona.